Automattic/jetpack
# Search: Migrate some widget JS to a static file

#### Changes proposed in this Pull Request:

* Move static JS to its own file so that it can be cached by the browser and possibly concatenated.
* Only load the JS/CSS if the widget is active.
* Rename admin JS to have -admin suffix.
* Remove inline HTML comment in favor of PHPDoc.

#### Testing instructions:

* Verify that the checkboxes on the front end still work.
* Verify that the JavaScript doesn't load when the widget isn't being used (`js/search-widget-filters.js` on the front end).
* Verify that the widget configuration in the customizer still works.
